### PR TITLE
[Fix](local merge) Fix local exchange dependencies acquired by local merge

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

LocalMergeExchangeSharedState is not used correctly and its dependencies are not used by source operators.

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/pipeline/local_exchange/local_exchanger.cpp [hunk 1]
```diff
@@ -84,7 +84,7 @@ bool Exchanger<BlockType>::_dequeue_data(LocalExchangeSourceLocalState& local_st
     } else if (all_finished) {
         *eos = true;
     } else {
-        std::unique_lock l(_m);
+        std::unique_lock l(get_mutex(channel_id));
```
Comment (yiguolei) [inline be/src/pipeline/local_exchange/local_exchanger.cpp:87]: sink(10) --> 10 个queue ---> 7 block

source --> 10 block --> 400 rows

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Review by Mryange — APPROVED** [review]

**Pull request merged**